pkg/execution/queue/queue.go

```diff
@@ -0,0 +1,25 @@
+package queue
+
+import (
+	"context"
+	"time"
+)
+
+type Queue interface {
+	Enqueuer
+	Worker
+}
+
+type Enqueuer interface {
```

Comment:
Should we call these `Producer` and `Consumer` instead of `Enqueuer` and `Worker` maybe to use more standard queue terms. For code readability I think it would be useful. `Enqueue` could also become `Send`.

Reply:
Done - thanks!